lib/cadet/courses/courses.ex

```diff
@@ -7,8 +7,124 @@ defmodule Cadet.Courses do
 
   import Ecto.Query
 
-  alias Cadet.Accounts.User
-  alias Cadet.Courses.{Group, Sourcecast, SourcecastUpload}
+  alias Cadet.Accounts.CourseRegistration
+
+  alias Cadet.Courses.{
+    AssessmentConfig,
+    AssessmentType,
+    Course,
+    Group,
+    Sourcecast,
+    SourcecastUpload
+  }
+
+  @doc """
+  Returns the course configuration for the specified course.
+  """
+  @spec get_course_config(integer) ::
+          {:ok, %Course{}} | {:error, {:bad_request, String.t()}}
+  def get_course_config(course_id) when is_ecto_id(course_id) do
+    case retrieve_course(course_id) do
+      nil ->
+        {:error, {:bad_request, "Invalid course id"}}
+
+      course ->
+        assessment_types =
+          AssessmentType
+          |> where(course_id: ^course_id)
+          |> Repo.all()
+          |> Enum.sort(&(&1.order < &2.order))
+          |> Enum.map(& &1.type)
+
+        {:ok, Map.put_new(course, :assessment_types, assessment_types)}
+    end
+  end
+
+  @doc """
+  Updates the general course configuration for the specified course
+  """
+  @spec update_course_config(integer, %{}) ::
+          {:ok, %Course{}} | {:error, {:bad_request, String.t()} | {:error, Ecto.Changeset.t()}}
+  def update_course_config(course_id, params) when is_ecto_id(course_id) do
+    case retrieve_course(course_id) do
+      nil ->
+        {:error, {:bad_request, "Invalid course id"}}
+
+      course ->
+        course
+        |> Course.changeset(params)
+        |> Repo.update()
+    end
+  end
+
+  defp retrieve_course(course_id) when is_ecto_id(course_id) do
+    Course
+    |> where(id: ^course_id)
+    |> Repo.one()
+  end
+
+  @doc """
+  Updates the assessment configuration for the specified course
+  """
+  @spec update_assessment_config(integer, integer, integer, integer, integer) ::
+          {:ok, %AssessmentConfig{}} | {:error, Ecto.Changeset.t()}
+  def update_assessment_config(course_id, order, early_xp, hours_before_decay, decay_rate) do
```

Comment:
Pass the properties as a map, otherwise if we get more properties this function is going to have more and more parameters